src/App.tsx

```diff
@@ -1,9 +1,34 @@
-function App() {
+import PanelsWrapper from './components/PanelsWrapper';
+import { useEffect, useState } from 'react';
+import { useConfig} from '@/contexts/ConfigContext'
+
+
+
+function App({customConfig}) {
+
+  const [loading, setLoading] = useState(true)
+  const { config, updateConfig} = useConfig()
+
+  useEffect(() => {
+    // load Config
+    // TODO: url config will need to get read
+    function loadConfig() {
+      const totalConfig = {...config, ...customConfig}
+      updateConfig(totalConfig)
+      setLoading(false)
+    }
+    loadConfig()
+  }, [])
```

Comment:
Shouldn't the dependencies array contain all variables it is using inside?

Reply:
I am not sure, when have a config change which affects only a specific component, then we would rerender the App component and its children, right ?  

Moreover, the loadConfig in App.tsx will be executed at the very first beginning.

Reply:
Actually I think that useEffect can make it possible to execute something only once. If I place code outside of useEffect it will be executed repeatedly. 

I removed useEffect() from PanelsWrapper, but to my opinion it seems like it is necessary in other components.

Reply:
Hm I still see the useEffect in PanelsWrapper. Also why do you need useEffects in other components?